docs/reference/cat/shards.asciidoc

```diff
@@ -34,6 +34,10 @@ For <<data-streams,data streams>>, the API returns information about the stream'
 `manage` <<privileges-list-cluster,cluster privilege>> to use this API.
 You must also have the `monitor` or `manage` <<privileges-list-indices,index privilege>>
 for any data stream, index, or alias you retrieve.
+Additionally, to retrieve shard information of restricted indices, you must have `allow_restricted_indices` set to `true`.
+Without the necessary <<roles-indices-priv,index privileges>>,
+you may encounter security exceptions when they are explicitly included in the `<target>` list,
+or receive incomplete shard results when using wildcards or when omitting the `<target>` parameter.
```

Comment:
I don't think we need to be this precise, or call out the restricted indices in this doc. 

I think a very minor addition to the existing text is sufficient: "to view the full information"

```
If the {es} {security-features} are enabled, you must have the `monitor` or
`manage` <<privileges-list-cluster,cluster privilege>> to use this API.
You must also have the `monitor` or `manage` <<privileges-list-indices,index privilege>>
to view the full information for any data stream, index, or alias you retrieve.
```

The docs end up being a pseudo contract, if they are very specific it can be hard to adjust implementation details as needed.

Reply:
> The docs end up being a pseudo contract, if they are very specific it can be hard to adjust implementation details as needed.

I was wondering if we should be more precise, but that makes sense. I like the proposal.

Reply:
pushed https://github.com/elastic/elasticsearch/pull/115650/commits/acb2ea267571380b8c606e9475bf0ae6397afdcb